python/hail/methods/io.py

```diff
@@ -1,18 +1,114 @@
 from hail.typecheck import *
 from hail.utils.java import Env, handle_py4j, joption
-from hail.api2 import MatrixTable
+from hail.api2 import Table, MatrixTable
 from hail.history import *
 from hail.expr.types import *
 from .misc import require_biallelic
 
+@handle_py4j
+@typecheck(table=Table,
+           address=strlike,
+           keyspace=strlike,
+           table_name=strlike,
+           block_size=integral,
+           rate=integral)
+def export_cassandra(table, address, keyspace, table_name, block_size=100, rate=1000):
+    """Export to Cassandra.
+
+    Warning
+    -------
+    :meth:`export_cassandra` is EXPERIMENTAL.
+    """
+
+    table._jkt.exportCassandra(address, keyspace, table_name, block_size, rate)
+
+@handle_py4j
+@require_biallelic
+@typecheck(dataset=MatrixTable,
+           output=strlike,
+           precision=integral)
+def export_gen(dataset, output, precision=4):
+    """Export variant dataset as GEN and SAMPLE files.
+
+    .. include:: ../_templates/req_tvariant.rst
+
+    .. include:: ../_templates/req_biallelic.rst
+
+    Examples
+    --------
+    Import genotype probability data, filter variants based on INFO score, and
+    export data to a GEN and SAMPLE file:
+    
+    >>> dataset = hc.import_gen('data/example.gen', sample_file='data/example.sample')
+    >>> dataset = dataset.filter_rows(agg.infoScore(dataset.GP).score >= 0.9) # doctest: +SKIP
+    >>> methods.export_gen(dataset, 'output/infoscore_filtered')
+
+    Notes
+    -----
+    Writes out the dataset to a GEN and SAMPLE fileset in the
+    `Oxford spec <http://www.stats.ox.ac.uk/%7Emarchini/software/gwas/file_format.html>`__.
+
+    This method requires a `GP` (genotype probabilities) entry field of type
+    ``Array[Float64]``. The values at indices 0, 1, and 2 are exported as the
+    probabilities of homozygous reference, heterozygous, and homozygous variant,
+    respectively. Missing `GP` values are exported as ``0 0 0``.
+
+    The first six columns of the GEN file are as follows:
+
+    - chromosome (`v.contig`)
+    - variant ID (`varid` if defined, else Contig:Position:Ref:Alt)
+    - rsID (`rsid` if defined, else ``.``)
+    - position (`v.start`)
+    - reference allele (`v.ref`)
+    - alternate allele (`v.alt`)
+
+    The SAMPLE file has three columns:
+
+    - ID_1 and ID_2 are identical and set to the sample ID (`s`).
+    - The third column (``missing``) is set to 0 for all samples.
+
+    Parameters
+    ----------
+    dataset : :class:`.MatrixTable`
+        Dataset with entry field `GP` of type Array[TFloat64].
+    output : :obj:`str`
+        Output file root for GEN and SAMPLE files.
+    precision : int
```

Comment:
It looks odd to me having `str` in the `:obj:` markup but not int. But I don't think we've agreed on a style.